source/fundamentals/object-serialization.txt

```diff
@@ -0,0 +1,202 @@
+====================
+Object Serialization
+====================
+
+.. default-domain:: mongodb
+
+Overview
+--------
+
+MongoDB stores data as :manual:`MongoDB Extended JSON
+<reference/mongodb-extended-json/>` containing fields that correspond
+with BSON data types. **Object serialization** instructs the driver how to
+translate data between Extended JSON and Java objects. The driver uses
+objects called **codecs** to describe object serialization.
```

Comment:
Unfortunately `Codec` is just an interface, while `CodecRegistry` is a full-blown class of its own that the reader will almost certainly have to use if they touch codecs at all in their code. To make matters even more complicated, there are additional interfaces like `CollectibleCodec` that you can optionally implement as well. Originally all mentions of `Codec` were monospaced like the interface name, but due to issues with pluralization and the fact that it's an interface, not a class you actually use, I decided to switch it to "codec".

I didn't explain the methods an implementation requires because i cover that detail in "create a custom codec" by linking to the codec javadoc, but if you feel strongly about it I can add more detail about the methods in that section.

Reply:
I'll leave it up to you on whether you want to term or class formatting for interfaces, but I prefer the latter. In either case, I'd request that you explain what a `Codec` is and what the connection is between that and the term codec to the user somewhere.

Reply:
Turning this into a request to document what a `Codec` is. We mention these in Usage Examples when applicable, e.g.: https://github.com/mongodb/docs-java/blob/master/source/usage-examples/distinct.txt#L32

Reply:
We currently explain the codec interface here:
https://github.com/mongodb/docs-java/pull/30/files#diff-b2d66599f35b6fa9dcf2b64f05957d8cR85
Because of the number of plural "codecs" occurrences, I'm not going to monospace the term -- unfortunately, there is still no good way to add a non-monospace "s" to the end of a monospace term that I'm aware of in rst.

Adding a bit about the ``Codec`` interface in the intro to make the relationship clearer as per your suggestion.